linode/nb/framework_resource.go

```diff
@@ -0,0 +1,241 @@
+package nb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+var _ resource.ResourceWithUpgradeState = &Resource{}
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			"linode_nodebalancer",
+			frameworkResourceSchema,
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data NodebalancerModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clientConnThrottle := int(data.ClientConnThrottle.ValueInt64())
+	createOpts := linodego.NodeBalancerCreateOptions{
+		Region:             data.Region.ValueString(),
+		Label:              data.Label.ValueStringPointer(),
+		ClientConnThrottle: &clientConnThrottle,
+	}
+
+	if !data.Tags.IsNull() {
+		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	nodebalancer, err := client.CreateNodeBalancer(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating a Linode NodeBalancer",
+			err.Error(),
+		)
+		return
+	}
+
+	data.ID = types.Int64Value(int64(nodebalancer.ID))
+	resp.Diagnostics.Append(data.parseNodeBalancer(ctx, nodebalancer)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
```

Comment:
> any attribute that does not have Computed set to true cannot be influenced by the provider; it can only be changed via the config or via state refreshing.

https://developer.hashicorp.com/terraform/plugin/framework/handling-data/schemas#computed

We are required not to change user configured (non-computed) values except in state refreshing (Read function). So we will have to separate parse computed and non-computed attributes functions.

Reply:
The language used by HashiCorp is a bit ambiguous in meaning, but it seems to imply that non-computed attributes should only be updated during read like you're saying. That said, it seems a bit odd to arbitrarily split non-computed and computed attributes into separate parse functions. I wonder if there's a better way to approach this.